publishing.qmd

```diff
@@ -1,5 +1,319 @@
+
+When you publish, Quarto creates a special `gh-pages` branch, renders your document or project to this branch, and pushes this branch to GitHub.
+This branch is then processed on GitHub's side, making it available at a URL of the form `{ user }.github.io/repo-name`.
+
+If you use GitHub through one of its "Free" plans you can only use GitHub Pages on public repositories.
+If you'd prefer your source documents weren't available publicly you'll either need to upgrade your GitHub plan, or use something like Netlify or Quarto Pub instead.
+
+You can read more about publishing to GitHub with `quarto publish` on the [GitHub Pages page](https://quarto.org/docs/publishing/github-pages.html#publish-command) in the Quarto documentation.
+
+An alternative to using `quarto publish` is to commit your rendered document or site to your repository in a `docs/` folder and set GitHub Pages to publish from there.
+You can read more about this workflow at [Render to `docs`](https://quarto.org/docs/publishing/github-pages.html#render-to-docs) on the Quarto documentation website .
+
+### Netlify
+
+``` {.bash filename="Terminal"}
+quarto publish netlify document.qmd
+```
+
+Netlify provides services for building, deploying, and managing websites.
+When you publish to Netlify using `quarto publish`, your document or site will be set up as what Netlify calls a [manual deploy](https://docs.netlify.com/cli/get-started/#manual-deploys) — you deploy by *manually* running `quarto publish`.
+Your output files are sent to Netlify and Netlify serves them at a URL of the form `{ random identifier }.netlify.app`.
+
+You can manage your deployed site via Netlify's website, and Netlify includes some neat features like one-click rollbacks to a previously deployed site, and the ability to use a custom domain.
+Netlify's free plans limit how many sites you can have and how much bandwidth is used to serve them.
+However, these limits will likely not be reached for low-traffic personal sites.
+
+Rather than using `quarto publish` you can get Netlify to deploy from Git provider.
+In this workflow, you commit your rendered document or site to your repsoitory in a folder (e.g. `docs/` or `_site`) and connect Netlify to your Git provider, setting the folder as the **Publish** directory.
+You can read more about this on the Quarto documentation website at [Publish from a Git Provider](https://quarto.org/docs/publishing/netlify.html#publish-from-git-provider).
+
+Netlify excels in continuous integration and you can set it up to preview deployments based on branches or pull requests.
+The features require a more complicated publishing set up you'll touch on in @sec-ci.
+
+### Other Providers
+
+You are most likely to use Posit Connect or Confluence in organizational settings where your IT department will manage them.
+Hugging Face has a more complicated set up and probably only makes sense if you are using Hugging Face beyond Quarto already.
```

Comment:
I wonder if we should link to homepages of these services like Netlify, Hugging Face, etc. at the first mention, for readers who may have never heard about them?